wpimath/src/main/java/edu/wpi/first/math/geometry/Pose2d.java

```diff
@@ -238,6 +241,26 @@ public Twist2d log(Pose2d end) {
     return new Twist2d(translationPart.getX(), translationPart.getY(), dtheta);
   }
 
+  /**
+   * Returns the nearest Pose2d from a list of poses. If two or more poses in the list have the same
+   * distance from this pose, return the one with the closest rotation component.
+   *
+   * @param poses The list of poses to find the nearest.
+   * @return The nearest Pose2d from the list.
+   */
+  public Pose2d nearest(List<Pose2d> poses) {
+    return Collections.min(
+        poses,
+        Comparator.comparing(
+                (Pose2d other) -> {
+                  System.out.println(this.getTranslation().getDistance(other.getTranslation()));
```

Comment:
I assume this is a debug print that should be removed?

Reply:
yep, thanks